### ISSUE TITLE
Sun and moon moving step by step

### ISSUE THREAD

**Issue by stevd3v** [CosmicDan-Minecraft/SimpleDayLengthExtender#2]
The sun and moon (provably the whole sky) are stuttering kind of like they are moving step by step when using low slowing rate.
A 2x slower it’s a very regular and annoying step by step, at 1.5 the pattern is less regular but also very noticeable.
I tried a 9x and it’s moving so slow that I think we can’t see the stutter.
I know this is certainly a limitation of the method used, but I was wondering if it is possible to implement so kind of interpolation or make use a different pattern to make it less noticeable with this kind of rates

**Comment by cosmicdan** [comment]
Have you made sure to have the mod on both client and server?

On 8 December 2024 10:28:56 am AEDT, Steven Renaud ***@***.***> wrote:
>The sun and moon (provably the whole sky) are stuttering kind of like they are moving step by step when using low slowing rate.
>A 2x slower it’s a very regular and annoying step by step, at 1.5 the pattern is less regular but also very noticeable.
>I tried a 9x and it’s moving so slow that I think we can’t see the stutter.
>I know this is certainly a limitation of the method used, but I was wondering if it is possible to implement so kind of interpolation or make use a different pattern to make it less noticeable with this kind of rates
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/CosmicDan-Minecraft/SimpleDayLengthExtender/issues/2
>You are receiving this because you are subscribed to this thread.
>
>Message ID: ***@***.***>
Regards,
Daniel Connolly

**Comment by stevd3v** [comment]
The issue is in single player, so if i’m not mistaken I I only have to install it in the mod folder

**Comment by cosmicdan** [comment]
Ok, you said "we" so I assumed multiplayer.

I don't really understand the issue to be honest, if you could make a video that would be great. But I think what you're describing is just normal behaviour of Minecraft. Even without the mod installed or set to default 1.0x, you can still see the sun or moon moving "step by step", no?

On 8 December 2024 4:58:28 pm AEDT, Steven Renaud ***@***.***> wrote:
>The issue is in single player, so if i’m not mistaken I I only have to install it in the mod folder
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/CosmicDan-Minecraft/SimpleDayLengthExtender/issues/2#issuecomment-2525433726
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
Regards,
Daniel Connolly

**Comment by stevd3v** [comment]
Sorry for the bad wording, you can see that on 1x everything is smooth except that we are able to see it move pixel by pixel when looking straight at it, but when slowing the day it starts stuttering with a certain pattern.
I'm on an m1 mac aarch64 fresh instance of minecraft with only your mod and embeddium.
The most annoying thing is that the stars are also doing the same thing, so it is very noticable at night.

https://github.com/user-attachments/assets/2411f593-c29e-4ebb-a6ae-cd5e0654a787

https://github.com/user-attachments/assets/eaf9d6aa-62f5-4d06-b6af-6691a3784c23

https://github.com/user-attachments/assets/d15fbd9f-b9c6-4de2-94ad-d131ccc4ffbe

**Comment by cosmicdan** [comment]
Does it persist if you remove Embeddium?

On 8 December 2024 7:04:49 pm AEDT, Steven Renaud ***@***.***> wrote:
>Sorry for the bad wording, you can see that on 1x everything is smooth except that we are able to see it move pixel by pixel when looking straight at it, but when slowing the day it starts stuttering with a certain pattern.
>I'm on an m1 mac aarch64 fresh instance of minecraft with only your mod and embeddium.
>
>https://github.com/user-attachments/assets/2411f593-c29e-4ebb-a6ae-cd5e0654a787
>
>https://github.com/user-attachments/assets/eaf9d6aa-62f5-4d06-b6af-6691a3784c23
>
>https://github.com/user-attachments/assets/d15fbd9f-b9c6-4de2-94ad-d131ccc4ffbe
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/CosmicDan-Minecraft/SimpleDayLengthExtender/issues/2#issuecomment-2525523109
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
Regards,
Daniel Connolly

**Comment by stevd3v** [comment]
Yes, I tried with your mod as the only one enabled and the same thing happened

**Comment by cosmicdan** [comment]
Fair enough, I'll see if I can improve it.

On 8 December 2024 7:21:26 pm AEDT, Steven Renaud ***@***.***> wrote:
>Yes, I tried with your mod as the only one enabled and the same thing happened
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/CosmicDan-Minecraft/SimpleDayLengthExtender/issues/2#issuecomment-2525536447
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
Regards,
Daniel Connolly

**Comment by stevd3v** [comment]
Thanks, I randomly noticed that if you set time to night it's all smooth until the moon raise up a bit, so this in addition to the different patterns at different speeds this could help locate the issue

**Comment by cosmicdan** [comment]
Fixed in 1.2.0. Sorry for the delay, better late than never 😅
